### PR TITLE
Fix warnings

### PR DESCRIPTION
**Pull request by Gurnaik-Randhawa**
Fixed warnings to improve code quality.

### CODE REVIEW

File: DFC.Digital/DFC.Digital.Data/Interfaces/IWebAppContext.cs [hunk 1]
```diff
@@ -24,7 +24,7 @@ public interface IWebAppContext
 
         bool IsUserAdministrator { get; }
 
-        NameValueCollection RequestQueryString { get; }
+        NameValueCollection RequestQueryString { get; set; }
```
Comment (Muthuramana) [inline DFC.Digital/DFC.Digital.Data/Interfaces/IWebAppContext.cs]: Is there a reason why we had to make it settable.
```suggestion
        NameValueCollection RequestQueryString { get; }
```

File: DFC.Digital/DFC.Digital.Web.Sitefinity.CmsExtensions.UnitTests/Controllers/ApprenticeshipVacancyReportControllerTests.cs [hunk 2]
```diff
@@ -73,13 +76,17 @@ public void IndexTest(int numberRecords, int numberOfApprenticeship)
...
 
-        public void IndexRedirectTest(int numberRecords, int numberOfApprenticeship)
+        [Fact]
+        public void IndexRedirectTest()
         {
             // Setup
-            A.CallTo(() => fakeReportRepository.GetJobProfileApprenticeshipVacancyReport()).Returns(GetDummyReportData(numberRecords, numberOfApprenticeship));
-            A.CallTo(() => fakeWebAppContext.RequestQueryString).Returns(null);
+            A.CallTo(() => fakeReportRepository.GetJobProfileApprenticeshipVacancyReport()).Returns(GetDummyReportData(1, 2));
+           // A.CallTo(() => fakeWebAppContext.RequestQueryString).Returns(null);
```
Comment (Muthuramana) [inline DFC.Digital/DFC.Digital.Web.Sitefinity.CmsExtensions.UnitTests/Controllers/ApprenticeshipVacancyReportControllerTests.cs]: Why can't we fake call to get property?
```suggestion
           query.Add("NOTctx", "something");
           A.CallTo(() => fakeWebAppContext.RequestQueryString).Returns(query);
```

File: DFC.Digital/DFC.Digital.Web.Sitefinity.Core/SitefinityExtensions/WebAppContext.cs [hunk 3]
```diff
@@ -51,7 +51,11 @@ public bool IsUserAdministrator
 
         public bool IsJobProfilePage => HttpContext.Current.Request.Url.ToString().ToLower().Contains("/job-profiles");
 
-        public NameValueCollection RequestQueryString => HttpContext.Current.Request.QueryString;
+        public NameValueCollection RequestQueryString
+        {
+            get { return HttpContext.Current.Request.QueryString; }
+            set { throw new NotImplementedException(); }
```
Comment (Muthuramana) [inline DFC.Digital/DFC.Digital.Web.Sitefinity.Core/SitefinityExtensions/WebAppContext.cs]:
```suggestion
            
```

Comment (Gurnaik-Randhawa) [reply DFC.Digital/DFC.Digital.Web.Sitefinity.Core/SitefinityExtensions/WebAppContext.cs]: Removed setter and faked get property.

**Pull request merged**